lxd/containers_get.go

```diff
@@ -77,11 +136,19 @@ func doContainersGet(d *Daemon, r *http.Request) (interface{}, error) {
 	// Parse the recursion field
 	recursionStr := r.FormValue("recursion")
 
+	// // Parse filter value
+	filterStr := r.FormValue("filter")
+
 	recursion, err := strconv.Atoi(recursionStr)
 	if err != nil {
 		recursion = 0
+
+		if filterStr != "" {
```

Comment:
That seems like a bad plan, we don't want to have the result type change based on the presence of a filter.

I'd rather we filter regardless of the recursion level. If recursion is 0, we'll need special handling, effectively going through most of the same path as recursion=1 but then only returning the URLs as is expected from recursion=0.

Obviously this should only be done if a filter is passed, if no filter is passed, recursion=0 should remain as quick as it is today.